### PR TITLE
Added a Sample Program to demonstrate use of supabase realtime-js with supabase-Auth

### PR DESCRIPTION
**Pull request by meera**
## What kind of change does this PR introduce?

Added a sample program - that shows a use case for real-time 'presence'. The sample program is built using NextJS and uses Supabase-Auth to get user email. You will see a list of users who have currently logged in and currently viewing the page.

## What is the current behavior?
We don't have sample programs for newly added features such as real-time presence and broadcast.

## What is the new behavior?
Sample program demonstrates supabase realtime-js functionality by tieing it with the supabase-auth library.

## Additional context
See previous pull request. 
https://github.com/supabase/realtime-js/pull/204

### CODE REVIEW

**Review by kiwicopple — APPROVED** [review]

**Review by kiwicopple — APPROVED** [review]

File: examples/realtime/nextjs-auth-presence/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+### 5. Create a .env.local file
+
+Create a .env.local file and add following environment variables.
+
+ ```
+ NEXT_PUBLIC_SUPABASE_URL=<<insert-your-db-url-here>>
+
+NEXT_PUBLIC_SUPABASE_ANON_KEY=<<insert-your-anon-key-here>> 
+```
```
Comment (kiwicopple) [inline examples/realtime/nextjs-auth-presence/README.md:46]: need formatting

**Review by kiwicopple — COMMENTED** [review]
Hey @meera 

I like the idea of a minimal example. it looks like this needs some formatting. Could you please run prettier over all the files?

**Review by kiwicopple — CHANGES_REQUESTED** [review]
Hey @meera

I like the idea of a minimal example. it looks like this needs some formatting. Could you please run prettier over all the files?

File: examples/realtime/nextjs-auth-presence/pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  supabaseClient.auth.onAuthStateChange((event, session) => {
+    switch( event) 
+    { case 'SIGNED_IN': 
+        router.push('/');
+        return;
+      case 'SIGNED_OUT':
+        router.push('/login');
+        return;
+    }
+  })
```
Comment (alaister) [inline examples/realtime/nextjs-auth-presence/pages/_app.tsx:23]: This needs wrapping in a `useEffect`. For example:
```javascript
useEffect(() => {
  const {
    data: { subscription },
  } = supabaseClient.auth.onAuthStateChange((event, session) => {
    switch (event) {
      case 'SIGNED_IN':
        router.push('/');
        return;
      case 'SIGNED_OUT':
        router.push('/login');
        return;
    }
  });

  return subscription.unsubscribe;
}, []);
```

**Review by alaister — CHANGES_REQUESTED** [review]
Hey @meera,

In addition to the formatting, I've left a comment regarding your use of `onAuthStateChange`.

You might also like to add unsubscribe calls to your other `useEffect` statements when you're subscribing to the presence channels.

Thank you!

File: examples/realtime/nextjs-auth-presence/pages/_app.tsx (hunk 2 above)
Comment (meera) [reply examples/realtime/nextjs-auth-presence/pages/_app.tsx]: Fixed. Wrapped onAuthStateChange inside useEffect

**Comment by meera** [comment]
Reformated files using prettier & have added useEffect.

**Review by alaister — APPROVED** [review]
Thanks, @meera, this looks great!
Please note I have replaced the deprecated `withPageAuth` as per the [docs here](https://github.com/supabase/auth-helpers/blob/main/packages/nextjs/MIGRATION_GUIDE.md#withpageauth-deprecated) and cleaned up the imports.

**Review by kiwicopple — APPROVED** [review]

**Pull request merged**